csv-report - script, pipeline, samples

felickz opened the pull request:
Script and sample pipeline for querying the GHAzDO alert APIs and uploading a CSV report of an org's alerts for each repo as an artifact.

[Readme.md](https://github.com/microsoft/GHAzDO-Resources/blob/e7a702376423864752a27cc280af811b79a96730/src/csv-report/README.md)


See: [Sample Report](https://github.com/microsoft/GHAzDO-Resources/blob/ca3318bf7ccd02edd4a8ef72ec66081f0b162298/src/csv-report/ghazdo-report-20230905.1.csv)

felickz commented on src/csv-report/ghazdo-csv-report.ps1:
```diff
@@ -0,0 +1,151 @@
+    Write-Host "##vso[artifact.upload artifactname=AdvancedSecurityReport]${env:BUILD_SOURCESDIRECTORY}/$reportName"
+}
+
+Write-Host "##vso[task.complete result=Succeeded;]DONE"
```
uses Write-Host. to output to Pipelines.

ncouraud commented on src/csv-report/ghazdo-csv-report.ps1:
```diff
@@ -0,0 +1,151 @@
+}
+
+# get list of projects in the Organization
+$url = "https://dev.azure.com/{0}/_apis/projects?api-version=6.0" -f $orgName
```
Is there a reason to not use the latest API versions?

ncouraud commented on src/csv-report/ghazdo-csv-report.ps1:
```diff
@@ -0,0 +1,151 @@
+    $repositoryId = $scan.RepoId
+    $alerts = $null
+    $parsedAlerts = $null
+    $url = "https://advsec.dev.azure.com/{0}/{1}/_apis/AdvancedSecurity/Repositories/{2}/alerts?useDatabaseProvider=true" -f $orgName, $project, $repositoryId
```
the useDatabaseProvider flag has been deprecated, and we shouldn't need it at this point.

ncouraud commented on src/csv-report/ghazdo-csv-report.ps1:
```diff
@@ -0,0 +1,151 @@
+    $parsedAlerts = $null
+    $url = "https://advsec.dev.azure.com/{0}/{1}/_apis/AdvancedSecurity/Repositories/{2}/alerts?useDatabaseProvider=true" -f $orgName, $project, $repositoryId
+
+    # Send out warnings for any org/project/repo that we cannot access alerts for!
```
Should we rather just call the enablement API to determine if AdvSec is enabled?

felickz replied:
Added ✅ ! 

I like the idea that you could use the debug logs of this script to gather info about which repos are enabled or not

ncouraud reviewed (commented):
I think the one metacomment I might make is whether we should consider any changes to improve support for running both in and outside of pipelines. 

In a similar vein - would it make sense to support passing in a scope to grab alerts from via Param vs. the inline variable?

felickz commented:
> I think the one metacomment I might make is whether we should consider any changes to improve support for running both in and outside of pipelines.
> 
> In a similar vein - would it make sense to support passing in a scope to grab alerts from via Param vs. the inline variable?

Script started from another template that was meant for AzDO pipeline, but i have modified it so you can run a bit easier with params.  Added a `scope` param to control org/proj/repo reporting
